pkg/api-server/testdata/inspect_dataplane_rules.golden.json

```diff
@@ -64,7 +64,7 @@
    },
    "policyType": "MeshTrafficPermission",
    "subset": {
-    "kuma.io/service": "client",
+    "__resource-name__": "client",
```

Comment:
This is a breaking change to the inspect api. I don't think it's acceptable

Reply:
Is it really? Looking at the code: 
```
type RuleInspectEntry struct {
  ...  
  Subset     map[string]string       `json:"subset"`
}
```

Subset is a map, so there could be anything. 

Other question is if we or anyone is using this value somewhere. In GUI there will be: 
<img width="1018" alt="image" src="https://github.com/kumahq/kuma/assets/9156357/5a1f9b1d-1101-4bb9-8471-6fa1af15f1e6">
Which looks weird. On the other side, when using MeshService we don't use `kuma.io/service` tag, but you will see this in matching. I can change it back to `kuma.io/service` but this will be confusing when using MeshService

The problem and why I've changed it is that in code we never know if you want to target old MeshService or new MeshService resource in policy. As both look the same in targetRef:
```
  to:
    - targetRef:
        kind: MeshService
        name: redis
```